### PR TITLE
Jambonz/privacy terms

### PR DESCRIPTION
**Pull request by love1119**

### CODE REVIEW

**Comment by love1119** [comment]
Hi, @kitajchuk 
Could you please review this PR when you get a chance?

**Comment by kitajchuk** [comment]
@love1119 Here's some feedback on this PR:

I see you've set up a reusable text layout component which is a step in the right direction. However, looking at the SASS it seems like a bit of a `copy+paste` from the `docs` SASS and then modified from there? I'm wondering if we should talk through a way to really normalize this format of raw `html` rendering and it's adjacent styles? In my mind we ought to be able to share the raw `html` styling between docs/terms/privacy, ideally, 100%. This would require a bit of a refactor as we'd need to set up the code to reflect this and adjust how the docs raw `html` gets rendered+styled as well. Since you've already moved in that direction, here's what I'd propose you do to reflect the change at large:

* Move the TextLayout component into the `jambonz-ui` library
* Use the TextLayout component in the Docs page component
* Normalize the styles (I'd just rename the docs partial and keep that vs what you have here for styles)

Let's see where that gets us? Rather than using the negative margins to overcome the padding for the Hero components on each legal page, I'd adjust the padding of the Hero targets in the `layout` SASS file. This is actually the reason they each also get a unique `className` since we already have scenarios on the main site where we tweak specific Hero styling. You can reference there in the `layout` SASS [here](https://github.com/jambonz/next-static-site/blob/main/styles/_layout.scss#L29).

I think this gets us pretty far along, but there's still one thing that's not sitting right with me...

Here you have some of the page data for these pages in the `yaml` data and some in the `docs` markdown data. This makes no sense (outside the fact I understand you were trying to leverage utilities already in place for rendering markdown). These legal pages are NOT developer docs and thus should not be located within the docs markdown folder. As it stands there is no conflict because the docs sidebar nav is controlled by a `yaml` data file so these pages don't show up. But, what if it was all 100% dynamic? These pages would show up in the docs sidbar nav which would be bad. Also, they are in the `rest` sub folder which also really makes no sense. These are legal pages, not developer docs about a REST API.

So, unfortunately this puts us in a weird position because the utilities need to be refactored to support loading markdown from locations other than the `docs` folder. One approach would be to try switching the private `_getDocs` function in `lib/data.js` to a more generic name like `getHtmlFromMarkdown` and making it publicly exported. Then this method could be used in a more ad hoc  manner for this kind of specific scenario. Since this method takes in the `filePath` you could just use it right there in your `getStaticProps` method for the page components. That might work.

Okay, this is a lot. I've pushed some more updates to the `main` branch so feel free to rebase those into your branch. I've moved all the typography into reusable mixins and added notes on adhering to typographic standards to the readme.

I'm going to make the change necessary here to the `lib/data.js` file and test that idea out. If it seems promising then you'll be able to rebase that as well and carry on as it were.

**Comment by kitajchuk** [comment]
@love1119 Go ahead and rebase main. I've updated `lib/data.js` to export the parser method so it can be used in page component files. You can see on this commit [here](https://github.com/jambonz/next-static-site/commit/b9bfad6eb509f5327920c248ed4ce39a5c761d0e).

My recommendation is to simply put the `privacy.md` and `terms.md` files in the pages directory right alongside their JS component counter parts (no worries, Next.js won't do anything with these). Then you can use this new public data method something like this in your page component JS:

```js
import Layout from '../components/layout';
import { Hero } from '../components/jambonz-ui';
import { getData, getParsedMarkdown } from '../lib/data';

// You'll need to load the nodejs path module
import path from 'path';

export default function Privacy({ data, parsed }) {
  return (
    <>
      <Layout siteData={data.site}>
        <Hero data={data.privacy.hero} subStyle="privacy" />
      </Layout>
      {/* do something with `parsed` */}
    </>
  );
}

export async function getStaticProps() {
  const data = getData('privacy');
  // Use the new public method in a more ad hoc manner like this...
  const parsed = await getParsedMarkdown(path.join(process.cwd(), 'pages', 'privacy.md'));

  return {
    props: {
      data,
      parsed,
    },
  };
}
```

**Review by kitajchuk — CHANGES_REQUESTED** [review]
Just checking the boxes as a reviewer here. I'll not say much here since I've already provided my feedback and collaboration efforts as comments above in the main thread. My requested changes are in those comments so get after it! :)

**Comment by love1119** [comment]
@kitajchuk 
Thank you for your details.
I got familiar with the structure of the project with your kindly help.
I will update PR soon.
Thanks a lot.

**Comment by love1119** [comment]
@kitajchuk 
I updated the PR. Could you please review this PR when you get a chance?

File: pages/terms.js [hunk 1]
```diff
@@ -1,21 +1,29 @@
 import Layout from '../components/layout';
-import { Hero } from '../components/jambonz-ui';
-import { getData } from '../lib/data';
+import { Hero, TextLayout } from '../components/jambonz-ui';
+import { getData, getParsedMarkdown } from '../lib/data';
 
-export default function Terms({ data }) {
+// You'll need to load the nodejs path module
```
Comment (kitajchuk) [inline pages/terms.js:5]: Overall this looks good. My only nit pick now would be that we don't need these comments (I just had those in my example for clarity sake). Same goes for `privacy.js`.

File: pages/terms.js [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
       <Hero data={data.terms.hero} subStyle="terms" />
+      <TextLayout data={parsed} />
     </Layout>
   );
 }
 
 export async function getStaticProps() {
   const data = getData('terms');
+  
+  // Use the new public method in a more ad hoc manner like this...
```
Comment (kitajchuk) [inline pages/terms.js:20]: Same nit pick here, we just don't need these comments specifically. Again, same goes for `privacy.js`.

**Comment by kitajchuk** [comment]
> @kitajchuk
> I updated the PR. Could you please review this PR when you get a chance?

Overall these changes look really good! I only had a few minor nit picks left so once those are resolved we're looking great!

**Comment by love1119** [comment]
@kitajchuk 
I just removed those comments.
Thank you.

**Comment by kitajchuk** [comment]
@davehorton @love1119 Too late now, and my bad for not bringing it up sooner (I didn't realize this was going to be merged right away like that), but for future reference I'd prefer if we `squash` our commits on PRs before merging. This way a feature comes into the `main` branch as a single, clean commit rather than all the piece meal commits on the feature branch. I have no issues with making as many commits as desired on a feature branch, but the `squash` is very desirable when merging back to `main`.

**Comment by davehorton** [comment]
gotcha, good point.

**Comment by kitajchuk** [comment]
@davehorton @love1119 I know I didn't do it on my original `docs-gen` branch so maybe I'm a hypocrite (it was early development in the experimental phases). I think as a project grows and more contributors come into the mix it begins to make more sense to keep the `main` branch history as clean as possible. I know squashing can be a bit of a pain when doing it manually. I'm not sure about Github, but I know Bitbucket lets you choose `squash` as a strategy for merges from the PR page which makes it nice and easy.

**Comment by love1119** [comment]
@kitajchuk 
I think `squash` is a good way to make main branch very clean.

**Pull request merged**